idea/src/org/jetbrains/kotlin/idea/intentions/ConvertFunctionToPropertyIntention.kt

```diff
@@ -88,8 +85,10 @@ class ConvertFunctionToPropertyIntention : SelfTargetingIntention<KtNamedFunctio
             function.setName(newName)
 
             val property = originalFunction.replace(psiFactory.createProperty(function.text)) as KtProperty
-            if (needsExplicitType) {
-                elementsToShorten.add(property.typeReference!!)
+
+            if (!hasExplicitType) {
+                val linebreak = property.getter?.node?.treePrev
```

Comment:
I think the reason for the current way is because it preserves annotations, modifiers, receiver type and other things I probably forgot. If we redid it, we would have to manually take care of it.